### DISCUSSION TITLE
Create new Standard Material on the fly

### DISCUSSION THREAD

**Discussion by jinincarnate** [brianzinn/react-babylonjs#234] [Q&A] [answered]
How do it create a new standard material when my model is loaded and apply to its meshes. The only examples i can find have a material already existing in the scene, but i need to create a new material when i need to

**Comment by jinincarnate** [comment, marked as answer]
Got it:

import { StandardMaterial } from '@babylonjs/core';
var testMat=new StandardMaterial("",scene);

**Comment by brianzinn** [comment]
That works.  You can also use a ref - like:
```
const matRef = useRef();
matRef.current = new StandardMaterial(...);
```

declarativly you could as well:
```
const matRef = useRef();

return (
  <>
     <standardMaterial ref={matRef} ... />
     <Model onModelLoaded={...} />
  </>
)
```
One advantage of above methods is that the material will likely be "ready" and loaded when the model is loaded, so not async waiting to apply the material.

**Reply by brianzinn** [reply]
be careful of one thing - the declarative `<StandardMaterial .../>` will attempt to attach to a mesh by walking down the tree graph! (ie: if a <ground> or another Mesh were found it would attach!  There is no opt-out mechanism on that behaviour.
